File: Algorithms/src/andy/level1/NumberStrToNumber.java
package andy.level1;

import java.util.Arrays;
import java.util.List;


public class NumberStrToNumber {
    public int solution(String s) {
        String str="";
        List<String> arr_str = Arrays.asList("zero", "one", "two", "three", "four", "five", "six", "seven", "eight", "nine");
        // String[] arr_str = {"zero", "one", "two", "three", "four", "five", "six", "seven", "eight", "nine"};
        
        String tmp="";
        for(int i=0;i<s.length();i++) {
            char c = s.charAt(i);           
            if(Character.isDigit(c) == true) 
                str += c;
            else {
                tmp += c;  // one
                // int idx = Arrays.asList(arr_str).indexOf(tmp);
                int idx = arr_str.indexOf(tmp);
                if(idx>=0) {
                    str += idx;
                    tmp="";                
                }                
            }
            
        }
        return Integer.parseInt(str);
    }
}

File: Andy/src/level1/javascript/NumberStrToNumber.js
// 숫자 문자열과 영단어
function solution(s) {
    var answer = 0;
    var arr = ["zero", "one", "two", "three", "four", "five", "six", "seven", "eight", "nine"];
    arr.forEach( function(d,i) { s = s.replace(new RegExp(d,"gi"),""+i); });
    return parseInt(s);
}